domain/src/DomainForm.php

```diff
@@ -80,27 +136,47 @@ public function form(array $form, FormStateInterface $form_state) {
       '#default_value' => $domain->isDefault(),
       '#description' => $this->t('If a URL request fails to match a domain record, the settings for this domain will be used. Only one domain can be default.'),
     );
-    $required = \Drupal::service('domain.validator')->getRequiredFields();
+    $form['homepage'] = [
```

Comment:
Two things I don't like here:

* Restricting the page to only nodes. (The last site I built has a View for a homepage.)

* This contradicts the existing logic that we have in Domain Config, which will cause confusion, because that will always present an option for overriding the homepage path (and has it's own Path Processor as well.)

My problem here is that this is a one-off approach to the problem of a UI, and I'd like to avoid that if possible.  We're also saving this as a config setting on the domain, not overriding the actual homepage setting, which I think will lead to confusion.

I can see a stronger case for perhaps adding a Domain Config UI that adds these form elements, but I would prefer that they be tied to the domain.config overrides, not the domain config entity itself. (Essentially we're mixing two systems here, which we should try to avoid.)